### PR TITLE
Ensure that all `TODO` and `FIXME` comments have associated issues

### PR DESCRIPTION
**Pull request by ahoppen**
I have found that standalone `FIXME` and `TODO` comments rarely get addressed and are thus not very valuable. 

Do one of the following for every `FIXME` or `TODO` comment
- Address them
- Add an issue that tracks the task
- Remove the comment if we are not planning to address it

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test n

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/BuildSystemIntegration/BuildSystemManager.swift [hunk 1]
```diff
@@ -382,7 +382,6 @@ extension BuildSystemManager: BuildSystemDelegate {
 }
 
 extension BuildSystemManager {
-  // FIXME: Consider debouncing/limiting this, seems to trigger often during a build.
```
Comment (hamishknight) [inline Sources/BuildSystemIntegration/BuildSystemManager.swift:385]: I guess this hasn't proved to be much of an issue?

File: Sources/LanguageServerProtocolJSONRPC/MessageSplitting.swift [hunk 2]
```diff
@@ -104,7 +104,6 @@ extension RandomAccessCollection where Element: Equatable {
       return nil
     }
 
-    // FIXME: use a better algorithm (e.g. Boyer-Moore-Horspool).
```
Comment (hamishknight) [inline Sources/LanguageServerProtocolJSONRPC/MessageSplitting.swift:107]: Could we use `firstRange(of:)` instead?

```swift
print((0..<10).firstRange(of: [4, 5, 6])!) // 4..<7
```

File: Tests/SourceKitLSPTests/SwiftCompletionTests.swift [hunk 3]
```diff
@@ -209,7 +209,6 @@ final class SwiftCompletionTests: XCTestCase {
       XCTAssertEqual(test.kind, .method)
       XCTAssertEqual(test.detail, "Void")
       XCTAssertEqual(test.filterText, "test(:)")
-      // FIXME:
```
Comment (hamishknight) [inline Tests/SourceKitLSPTests/SwiftCompletionTests.swift:212]: Best FIXME

**Review by hamishknight — APPROVED** [review]

File: Sources/BuildSystemIntegration/BuildSystemManager.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/BuildSystemIntegration/BuildSystemManager.swift:385]: I am not aware of any issue caused by not debouncing this in the last 5 years…

File: Sources/LanguageServerProtocolJSONRPC/MessageSplitting.swift (hunk 2 above)
Comment (ahoppen) [reply Sources/LanguageServerProtocolJSONRPC/MessageSplitting.swift:107]: I’ll try that in a follow-up PR.

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**